django/core/management/commands/makemessages.py

```diff
@@ -168,6 +168,17 @@ def normalize_eols(raw_contents):
     return '\n'.join(lines_list)
 
 
+def normalize_locale(locale_str, out):
+    """
+    Normalize locale string in different formats to use a single locale format.
+    Eg. (zh-cn, zh_cn, zh_CN, ZH-CN) all map to zh_CN
+    """
+    language = locale_str.lower().replace('_', '-')
+    if language != locale_str:
+        out.write('locale normalized to %s' % language)
```

Comment:
The log message makes no sense. You are trying to normalize (or rather kludge) something that should be a locale to be a language code/tag, to then convert to a locale. But you are saying the locale has been normalized to a language code/tag which is incorrect.

A language code/tag is not a locale, and here we *should be providing a locale*, hence my misgivings already stated about this whole thing.

Regardless, you probably want to do this:
```python
def normalize_locale(original, stdout):
    """
    Normalizes incorrect locale strings, e.g. zh-cn, zh_cn, ZH-CN are converted to zh_CN.
    """
    corrected = to_locale(original.lower().replace('_', '-'))
    if original != corrected:
        stdout.write('Normalized %s to %s.' % (original, corrected))
    return corrected
```